test/k4FWCoreTest/src/components/ExampleFunctionalConsumerMultiple.cpp

```diff
@@ -107,6 +114,15 @@ struct ExampleFunctionalConsumerMultiple final
             << tracks[0].getChi2() << ", " << tracks[0].getNdf();
       throw std::runtime_error(error.str());
     }
+
+    for (size_t j = 0; j < links.size(); j++) {
+      if (links[j].getFrom() != recos[j] || links[j].getTo() != particles[j]) {
+        std::stringstream error;
+        error << "Wrong data in links collection, link" << j << " expected " << recos[0] << ", " << particles[j]
+              << " got " << links[j].getFrom() << ", " << links[j].getTo();
```

Comment:
Doesn't need to happen in this PR (or maybe not at all?), but we are starting to get a wild mix of `fmt::format` and `std::stringstream` for formatting error messages. I think we should settle for one for Key4hep and then try to use that consistently.

Reply:
Yes, let's go for `std::format` once that Ubuntu 22 is dropped which will be soon since Gaudi hasn't been supporting GCC 11 for a while and it always has to be patched.

Reply:
I think we can go for `fmt` for now in any case. Gaudi uses that internally quite heavily, in that case we wouldn't need to rely on compiler support as much.

Reply:
But then we change it once to change it later to `std::format`, and if we were to follow the systems that are supported by Gaudi we would already have compiler support for it (they only build for GCC >= 13 and Clang >= 16, same as the LCG stacks) so I would perefer to keep the `std::stringstream`s and do the changes at once in the not so distant future.